### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by nobu**
Typos in a manual, example descriptions, a test method name and an already fixed old typo constant.

## What was the end-user or developer problem that led to this PR?

None.

## What is your fix for the problem, implemented in this PR?

Fixed.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [ ] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [ ] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

File: test/rubygems/test_gem_resolver.rb [hunk 1]
```diff
@@ -266,14 +266,14 @@ def test_resolve_remote_missing_dependency
 
     res = Gem::Resolver.new [a_dep], Gem::Resolver::IndexSet.new
 
-    e = assert_raise Gem::UnsatisfiableDepedencyError do
+    e = assert_raise Gem::UnsatisfiableDependencyError do
```
Comment (simi) [inline test/rubygems/test_gem_resolver.rb:269]: Is it time to remove the misspelled class as well?

https://github.com/rubygems/rubygems/blob/5b910cc4a78827c0571252dd93776b559cef608a/lib/rubygems/exceptions.rb#L283-L286

Comment (deivid-rodriguez) [reply test/rubygems/test_gem_resolver.rb:269]: I guess, yeah!

Comment (nobu) [reply test/rubygems/test_gem_resolver.rb:269]: Or warn first?
```ruby
Gem.deprecate_constant :UnsatisfiableDepedencyError
```

**Comment by deivid-rodriguez** [comment]
CI failed because man pages had gotten out of sync, but I fixed it, so this should automerge properly now.

File: test/rubygems/test_gem_resolver.rb (hunk 1 above)
Comment (deivid-rodriguez) [reply test/rubygems/test_gem_resolver.rb:269]: Even if it's a very old constant, I guess it's safest to deprecate it first. We can do it on a separate PR.

**Pull request merged**